python/cuml/thirdparty_adapters/adapters.py

```diff
@@ -285,82 +278,52 @@ def check_array(array, accept_sparse=False, accept_large_sparse=True,
         return X
 
 
-_input_type_to_str = {
-    numpyArray: 'numpy',
-    cupyArray: 'cupy',
-    cuSeries: 'series',
-    cuDataFrame: 'dataframe',
-    pdSeries: 'numpy',
-    pdDataFrame: 'numpy'
-}
-
-
-def get_input_type(input):
-    # function to access _input_to_str, while still using the correct
-    # numba check for a numba device_array
-    if type(input) in _input_type_to_str.keys():
-        return _input_type_to_str[type(input)]
-    elif numbaArray.is_cuda_ndarray(input):
-        return 'numba'
-    elif isinstance(input, cpu_sparse.csr_matrix):
-        return 'scipy_csr'
-    elif isinstance(input, cpu_sparse.csc_matrix):
-        return 'scipy_csc'
-    elif isinstance(input, cpu_sparse.coo_matrix):
-        return 'scipy_coo'
-    elif isinstance(input, gpu_sparse.csr_matrix):
-        return 'cupy_csr'
-    elif isinstance(input, gpu_sparse.csc_matrix):
-        return 'cupy_csc'
-    elif isinstance(input, gpu_sparse.coo_matrix):
-        return 'cupy_coo'
+def cuml_estimator(cuml_class):
+    from cuml._thirdparty.sklearn.utils.skl_dependencies import BaseEstimator
+
+    orig_init = cuml_class.__init__
+
+    def __init__(self, *args, **kwargs):
+        BaseEstimator.__init__(self, *args, **kwargs)
+        for param in ['handle', 'verbose', 'output_type']:
+            if param in kwargs:
+                del kwargs[param]
+        orig_init(self, *args, **kwargs)
+
+    cuml_class.__init__ = __init__
+
+    return cuml_class
```

Comment:
I have concerns about this decorator. You are making it initialize from sklearn's estimator via `BaseEstimator.__init__(self, *args, **kwargs)` but aren't adding the base estimator to the class inheritance. It's not clear why this would be useful since `issubclass` and `isinstance` wouldnt work with sklearn's classes.

Again, I think an offline discussion would be beneficial to understand the motivation for using a descriptor in this was as opposed to a metaclass or `__init__subclass__()`.

Reply:
The motivation behind the `cuml_estimator` is to provide the features of the `Base` class with minimal change to the sklearn side of the code. It allows the [initialization](https://github.com/rapidsai/cuml/blob/ff1f419ce75c28af5946a589fc50c1784332bd15/python/cuml/_thirdparty/sklearn/utils/skl_dependencies.py#L33) of the `Base` class with the `handle`, `verbose` and `output_type` parameters. This code is supposed to be specific to preprocessing and models are expected to already inherit from `BaseEstimator`.

It's true that an `__init__subclass__` method in the `BaseEstimator` class would probably be a preferable solution. Thanks for the tip.